### PR TITLE
fix: make inference_api handle None type

### PR DESCRIPTION
**Pull request by aarnphm**
- fix: ignore user_defined_callback if it is a lambda function
- fix: handling None type

### CODE REVIEW

**Review by sauyon — APPROVED** [review]
Looks fine, one comment if you want to.

File: src/bentoml/_internal/service/inference_api.py [hunk 1]
```diff
@@ -126,6 +127,9 @@ def __init__(
                                 f"Expected type of argument '{second_arg}' to be '{input_type}', got '{sig.parameters[second_arg].annotation}'"
                             )
 
+        self.name = name
+        self.func = user_defined_callback
```
[the remark this answers is not in the document]
Comment (aarnphm) [reply src/bentoml/_internal/service/inference_api.py]: added

File: src/bentoml/_internal/service/inference_api.py [hunk 2]
```diff
@@ -126,6 +127,17 @@ def __init__(
...
                             )
 
+        if user_defined_callback is not None:
+            self.func = user_defined_callback
+        else:
+
+            def nop(*args: t.Any, **kwargs: t.Any):
+                return
+
+            self.func = nop
```
Comment (sauyon) [inline src/bentoml/_internal/service/inference_api.py:137]: Was there a problem with just
```suggestion
        self.func = lambda _: None
```

Comment (aarnphm) [reply src/bentoml/_internal/service/inference_api.py:137]: nothing. Just preference.

**Review by sauyon — APPROVED** [review]
:+1:

**Pull request merged**